inference-engine/tests/unit/gna/ngraph/transformations/gna_split_convolution_with_large_buffer_size.cpp

```diff
@@ -0,0 +1,352 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include "transformations/split_convolution_with_large_buffer_size.hpp"
+
+#include "common_test_utils/ngraph_test_utils.hpp"
+#include <ngraph/function.hpp>
+#include <ngraph/opsets/opset7.hpp>
+#include <ngraph/pass/manager.hpp>
+#include <transformations/init_node_info.hpp>
+
+namespace testing {
+namespace {
+
+// use constexpr uint32_t bufferMaxSize = 65528 from gna_limitations
+
+struct SubGraph {
+    std::shared_ptr<ngraph::opset7::Parameter> input_node;
+    ngraph::OutputVector output_nodes;
+};
+
+SubGraph createSubGraphSolid(const ngraph::Shape& input_shape, const ngraph::Shape& kernel_shape) {
+    SubGraph sub_graph;
+
+    sub_graph.input_node = std::make_shared<ngraph::opset7::Parameter>(ngraph::element::f32,
+                                                                       input_shape);
+    auto kernel = ngraph::opset1::Constant::create(ngraph::element::f32,
+                                                   kernel_shape, {1});
+
+    auto convolution_operation = std::make_shared<ngraph::opset7::Convolution>(sub_graph.input_node,
+                                                                               kernel,
+                                                                               ngraph::Strides{1, 1},
+                                                                               ngraph::CoordinateDiff{0, 0},
+                                                                               ngraph::CoordinateDiff{0, 0},
+                                                                               ngraph::Strides{1, 1});
+    sub_graph.output_nodes.push_back(convolution_operation);
+
+    return sub_graph;
+}
+
+SubGraph createSubGraphSplitted() {
+    SubGraph sub_graph;
+
+    sub_graph.input_node = std::make_shared<ngraph::opset7::Parameter>(ngraph::element::f32,
+                                                                      ngraph::Shape{1, 64, 4096, 4096});
+
+    auto split_node_c1 = ngraph::opset7::Constant::create(ngraph::element::i64, ngraph::Shape({1}), std::vector<int64_t>{3});
+    auto split_node_c2 = ngraph::opset7::Constant::create(ngraph::element::i64, ngraph::Shape({5}), {960, 960, 960, 960, 256});
+    auto split_node = std::make_shared<ngraph::opset7::VariadicSplit>(sub_graph.input_node,
+                                                                      split_node_c1,
+                                                                      split_node_c2);
+
+    auto kernel = ngraph::opset1::Constant::create(ngraph::element::f32,
+                                                   ngraph::Shape{1, 64, 1, 1}, {1});
+
+    for (int i = 0; i < 5; ++i) {
```

Comment:
It does not seem to be universal. The number of splits depends on the input shape. I think this function should be parametrized as well.